### ISSUE TITLE
Monthly Competitive Coding Challenges (Online / Offline)

### ISSUE THREAD

**Issue by arijeetsat** [p-society/goals#22]
Discussion regarding the monthly coding challenges can be done here. Have to decide the platform, duration, competition type, etc. @bkhanale  Prateek Pattnaik @sr6033  @nawedx .

**Comment by palash25** [comment]
Please add  Prateek Pattnaik to the org I don't know his github handle

**Comment by palash25** [comment]
Monthly coding events can be done on online platforms like hackerearth codechef etc and for onsite events like swig and code and principium we can host them on our judge that we are currently developing it will be custom made to suit our requirements.

**Comment by palash25** [comment]
@bkhanale can you set up campus chapters for Hackerrank or codechef maybe we can get them as sponsors for Advaita if we maintain active chapters.

**Comment by palash25** [comment]
> We can arrange a Tech-Talk with Anup Kalbalia (Tech lead, CodeChef) if we want next year.

Yes please @bkhanale that would be great

**Comment by palash25** [comment]
Can anyone ask Swapnil Sourabh and Abhishek Patel to accept the invites and become active here we need all the help that we can.

**Comment by arijeetsat** [comment]
@bkhanale  Keep a watch at all other opportunities regarding competitive coding events, so that we can get max. incentives and resources as well as outreach.

**Comment by bkhanale** [comment]
There would be few demands from their side as well. They basically need us to cover the travelling expenses and accommodation costs. Though if he had planned to come to Bhubaneswar at that time then the costs would reduce.

**Comment by arijeetsat** [comment]
That part can be taken care by the Tech Club funds and all.

**Comment by palash25** [comment]
> That part can be taken care by the Tech Club funds and all.

We have those? Do we ever get to use them?

**Comment by shivank01** [comment]
@bkhanale @palash25 Inviting Anup Kalbalia would be great. We can also think about inviting Gaurav Sen. He has recently given Tech Talks on competitive programming in some NIT's.

**Comment by palash25** [comment]
See  #24

**Comment by sr6033** [comment]
I wanted to bring Anup this semester only but we fell short of budget. Make sure to keep a separate fund either from ACM or TECH for speakers from the beginning.

**Comment by palash25** [comment]
> I wanted to bring Anup this semester only but we fell short of budget.

For what do we have to spend money on? Is it airfare? Can't these speakers just buy their own flight tickets? I mean they are all working professionals.

**Comment by arijeetsat** [comment]
There are some terms these speakers follows , we have to talk to them before hand about that, as far as it is possible , hospitality wont be a problem from tech club side but travelling budget is something to think about.

**Comment by palash25** [comment]
I'm all for hospitality but I'm not in favor of calling speakers who work for well-established startups and large conglomerates but ask students to pay for their airfare. Its okay if these speakers are students themselves then we can think about paying for their tickets.

But that's just me sharing my opinion and if you guys unanimously decide to call a speaker and fund for their air travel I won't have any problem with that.

**Comment by arijeetsat** [comment]
@palash25 can you please ,merge the speaker discussion to one issue only. Kind of staggered ho raha hai.

**Comment by palash25** [comment]
Actually there is no need to merge the issue we just need to stop talking about tech talks on this thread. Not sure who started and deviated us from this issue, I've lost all context.

**Comment by arijeetsat** [comment]
**Update : The Coding Challenge for the month of May , successfully conducted. Codechef HEAT JAM May 2018**

**Comment by sonakshisaxena1** [comment]
Can we have the competitive coding challenges every fortnight? 
This will encourage people to learn and practice more. Also, it will help students to gain confidence regarding their coding skills which in turn will help them participate in more coding contests online.

**Comment by arijeetsat** [comment]
Every fortnight is a good idea but we need more people to come forward to manage it, right now Bhushan and Prateek are the problem setters, if we will plan for every fortnight we can have then on Hackerearth and Codechef alternatively ,but again the same thing more people to come forward and take responsibility . Also we can allow students from other colleges and keep it open. More thoughts required on this.

**Comment by sr6033** [comment]
@arijeetsat @bkhanale make a list of potential problem setters. And then we can alternatively ask them to set questions for the events.

I and @ankitjha97 can also help occasionally.

**Comment by bkhanale** [comment]
@sonakshisaxena1 Conducting contests on our own every fortnight would not be feasible. The most we can do is one contest per month. I have the list of problem setters with me, but certainly for getting one contest per month we will need more problem setters. Also, since Codeforces and Codechef have many short contests we can always promote them for participation. They have some quality problems and editorials as well.

**Comment by arijeetsat** [comment]
Okay one thing, what is our main purpose of holding contests, lets be clear abt then the frequency matters.
As per me, we are holding them to encourage our students to participate in local contest ,compete against each other , also to maintain a CC culture.

Thoughts please??

**Comment by palash25** [comment]
I agree with @bkhanale on this. You cannot maintain this kind of frequency of contests we just don't have that many setters.

Also after a while, CP newbies will move onto real competitions like Long challenges and the popularity of these challenges will go down.

**Comment by arijeetsat** [comment]
Monthly with good quality questions and more participation  is what we should focus on maintaing.👍👍

**Comment by bkhanale** [comment]
@arijeetsat Yes, but just not competing but improve themselves, learn new concepts. We need more good programmers for competitions like CodeJam, Hackercup, ACM ICPC. And the Codechef long and Codeforces rounds do good job in learning part. To improve participation we need the local contests.

**Comment by sr6033** [comment]
One monthly challenge by the club and we can give some goodies to good performers in other challenges going on in codechef, hackerrank, codeforces etc etc,

**Comment by sonakshisaxena1** [comment]
Okay. I get the issue of setting up the problems. 

But as @bkhanale said that there are many short contests available in Codechef/Codeforces, then can we gather up people and encourage them to give any selected contest together, in CG or some other rooms? 

This will stop them to procrastinate about giving a contest someday and will also help them to discuss solutions and editorials together. They might be able to form a good team for such contests like ICPC etc in future too.

**Comment by arijeetsat** [comment]
> help them to discuss solutions and editorials together. They might be able to form a good team for such contests like ICPC etc in future too.

This is something that has to be done, discussion on monthly challenges on the major platforms , at least Codechef, if not all. Also, for own competition, we also need to have editorials.
